doc/api/n-api.md

```diff
@@ -117,6 +117,61 @@ available to the module code.
 
 \* Indicates that the N-API version was released as experimental
 
+The N-APIs associated strictly with accessing ECMAScript features from native
+code can be found separately in `js_native_api.h` and `js_native_api_types.h`.
+The APIS defined in these headers are included in `node_api.h` and
+`node_api_types.h`. The separate headers exist to provide for the possibility of
+multiple implementations of N-API.
+
+The Node.js-specific parts of an addon can be separated from the code that
+exposes the actual functionality to the JavaScript environment so that the
+latter may be used with multiple implementations of N-API:
+
+```C
+// addon.h
+#ifndef _ADDON_H_
+#define _ADDON_H_
+#include <js_native_api.h>
+napi_value create_addon(napi_env env);
+#endif  // _ADDON_H_
+```
+
+```C
+// addon.c
+#include "addon.h"
+napi_value create_addon(napi_env env) {
+  napi_value result;
+  assert(napi_create_object(env, &result) == napi_ok);
+  napi_value exported_function;
+  assert(napi_create_function(env,
+                              "doSomethingUseful",
+                              NAPI_AUTO_LENGTH,
+                              DoSomethingUseful,
+                              NULL,
+                              &exported_function) == napi_ok);
+  assert(napi_set_named_property(env,
+                                 result,
+                                 "doSomethingUseful",
+                                 exported_function) == napi_ok);
+  return result;
+}
+```
+
+```C
+// addon_node.c
+#include <node_api.h>
+
+static napi_value Init(napi_env env, napi_value exports) {
+  return create_addon(env);
+}
+
+NAPI_MODULE(NODE_GYP_MODULE_NAME, Init)
+```
+
+With files arranged in this fashion it is possible to build the addon for
+Node.js, and also to reuse the code in `addon.c` and `addon.h` for a potential
+implementation of N-API outside of Node.js.
```

Comment:
Maybe move this to above the example and update to be:

For example, by using the js_xx headers files can be arranged  to build the addon for
Node.js, and also to reuse the code in `addon.c` and `addon.h` for an
implementation of N-API outside of Node.js.

Reply:
I elaborated on the role of `addon.c`, `addon.h`, and the motivation behind referring to `js_native_api.h` above the example.